### PR TITLE
Update StartupCopy.cs

### PR DESCRIPTION
**Pull request by qqbuby**
This property is obsolete and will be removed in a future version. The recommended alternative is `Microsoft.AspNetCore.Http.CookieBuilder.Name` on `Microsoft.AspNetCore.Builder.SessionOptions.Cookie`.

[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/fundamentals/app-state?branch=pr-en-us-4271)

### CODE REVIEW

**Comment by scottaddie** [comment]
@qqbuby This change will require an upgrade of the sample app to ASP.NET Core 2.0 before it will compile. Are you interested in doing that upgrade?

**Comment by qqbuby** [comment]
@scottaddie

Should we provide two versions, 1.0 and 2.0?

> This change will require an upgrade of the sample app to ASP.NET Core 2.0 before it will compile. Are you interested in doing that upgrade?

**Comment by scottaddie** [comment]
@qqbuby Yes, please.

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        const string SessionKeyName = "_Name";
+        const string SessionKeyYearsMember = "_YearsMember";
+        const string SessionKeyDate = "_Date";
+
+        public IActionResult Index()
+        {
+            // Requires using Microsoft.AspNetCore.Http;
+            HttpContext.Session.SetString(SessionKeyName, "Rick");
+            HttpContext.Session.SetInt32(SessionKeyYearsMember, 3);
+            return RedirectToAction("SessionNameYears");
```
Comment (scottaddie) [inline aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs]: In an effort to promote best practices, let's use the `nameof` operator here. The new line will look as follows:
```csharp
return RedirectToAction(nameof(SessionNameYears));
```

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            return Content($"Name: \"{name}\",  Membership years: \"{yearsMember}\"");
+        }
+        #endregion
+
+        #region snippet2
+        public IActionResult SetDate()
+        {
+            // Requires you add the Set extension method mentioned in the article.
+            HttpContext.Session.Set<DateTime>(SessionKeyDate, DateTime.Now);
+            return RedirectToAction("GetDate");
```
Comment (scottaddie) [inline aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs]: Use `nameof` here too.

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/WebAppSession.csproj [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <PreserveCompilationContext>true</PreserveCompilationContext>
+    <AssemblyName>WebAppSession</AssemblyName>
+    <OutputType>Exe</OutputType>
```
Comment (scottaddie) [inline aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/WebAppSession.csproj]: An output type of exe is implied by the Web SDK. This node can be removed.

**Comment by scottaddie** [comment]
@HaoK Please review

**Comment by HaoK** [comment]
@Tratcher do you own session?

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs (hunk 1 above)
Comment (qqbuby) [reply aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs]: fixed

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs (hunk 2 above)
Comment (qqbuby) [reply aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/Controllers/HomeController.cs]: fixed

File: aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/WebAppSession.csproj (hunk 3 above)
Comment (qqbuby) [reply aspnetcore/fundamentals/app-state/sample/src/WebAppSessionDotNetCore2.0App/WebAppSession.csproj]: fixed

**Comment by scottaddie** [comment]
@qqbuby Thank you for all your hard work and patience on this one! Your changes will go live within the next 24 hours.

**Pull request merged**